pkg/applier/manager.go

```diff
@@ -148,44 +149,53 @@ func (m *Manager) runWatchers(ctx context.Context) error {
 	}
 }
 
-func (m *Manager) createStack(ctx context.Context, name string) error {
+func (m *Manager) createStack(ctx context.Context, name string) {
 	// safeguard in case the fswatcher would trigger an event for an already existing stack
 	if _, ok := m.stacks[name]; ok {
-		return nil
-	}
-	m.log.WithField("stack", name).Info("registering new stack")
-	sa, err := NewStackApplier(ctx, name, m.KubeClientFactory)
-	if err != nil {
-		return err
+		return
 	}
 
+	stackCtx, cancelStack := context.WithCancel(ctx)
+	stack := stack{cancelStack, NewStackApplier(name, m.KubeClientFactory)}
+	m.stacks[name] = stack
+
 	go func() {
-		_ = sa.Start()
-	}()
+		log := m.log.WithField("stack", name)
+		for timer := time.NewTimer(0 * time.Second); ; timer.Reset(10 * time.Second) {
+			select {
+			case <-timer.C:
+				log.Info("Running stack")
+				if err := stack.Run(stackCtx); err != nil {
+					log.WithError(err).Error("Failed to run stack")
+				}
 
-	m.stacks[name] = sa
-	return nil
+			case <-stackCtx.Done():
+				timer.Stop()
+				log.Info("Stack done")
+				return
+			}
+		}
+	}()
 }
 
-func (m *Manager) removeStack(ctx context.Context, name string) error {
-	sa, ok := m.stacks[name]
-
+func (m *Manager) removeStack(ctx context.Context, name string) {
+	stack, ok := m.stacks[name]
 	if !ok {
 		m.log.
 			WithField("path", name).
 			Debug("attempted to remove non-existent stack, probably not a directory")
-		return nil
-	}
-	sa.Stop()
-	err := sa.DeleteStack(ctx)
-	if err != nil {
-		m.log.WithField("stack", name).WithError(err).Warn("failed to stop and delete a stack applier")
-		return err
+		return
 	}
-	m.log.WithField("stack", name).Info("stack deleted succesfully")
+
 	delete(m.stacks, name)
+	stack.CancelFunc()
 
-	return nil
+	log := m.log.WithField("stack", name)
```

Comment:
nitpick: more idiomatic is not inverse compassion and make a return in negative case. That flattens the code flow and keeps every unsuccessful code branched while keeping successful flow on the top level.

e.g.
```
if err := stack.DeleteStack(ctx); err != nil {
   log.WithError(err)
   return
}
// successful branc

Reply:
Done.